app.js

```diff
@@ -14,3 +14,96 @@ const books = [
   { author: "Aldous Huxley", title: "Animal Farm", id: 9 },
   { author: "Aldous Huxley", title: "Brave New World", id: 10 },
 ];
+
+class Patron {
+  constructor(name, id) {
+    this.name = name;
+    this.id = id;
+  }
+}
+
+class BookStatus {
+  constructor() {
+    this.checkedOut = false;
+    this.patron = null;
+  }
+
+}
+
+class Book {
+  constructor(title, author, id) {
+    this.title = title;
+    this.author = author;
+    this.id = id;
+    this.status = new BookStatus();
+  }
+}
+
+class Library {
+  constructor(name, books) {
+    this.name = name;
+    this.books = books;
+  }
+
+  addBook(title, author, id) {
+    this.books.push(new Book(title, author, id));
+  }
+
+  removeBook(id) {
+    this.books = this.books.filter((book) => book.id !== id);
+  }
+
+  getBookById(id) {
```

Comment:
Nice work, just something to add in the future, in case no book was found by an incorrect id, or if it was removed, you can add this:
```
  getBookById(id) {
    const foundBook = this.books.find((book) => book.id === id);

    if (!foundBook) {
      console.log(`No book found with the id of ${id}`);
      return null;
    }

    return foundBook;
  }

  checkOutBook(id, patron) {
    const foundBook = this.getBookById(id);

    if (!foundBook) {
      return;
    }

    const message = `${foundBook.title} has been checked out by ${patron.name}`;

    if (!foundBook.status) {
      foundBook.status = new BookStatus(true, patron);
      console.log(message);
      return;
    }

    if (foundBook.status.checkedOut) {
      message = "Book already checked out.";
      console.log(message);
      return;
    }

    foundBook.status.checkedOut = true;
    foundBook.status.patron = patron;

    console.log(message);
  }
  ```